### PR TITLE
Metrics Webapp 

### PR DESCRIPTION
**Pull request by mmsesay**
In this RP I worked on the following:

- Setup project files
- Configure linters and redux 
- Make API calls to fetch data from external API
- Restructured the data and dispatch it to the redux store
- Read from the store and display list of cities 
- Implemented dynamic single detail view for each city that is clicked
- Write unittest
- Design the application's UI

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello @mmsesay 

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }],
+    "import/no-named-as-default": 0,
+    "import/prefer-default-export": "off",
+    "import/no-cycle": "off",
+    "no-trailing-spaces": "off",
+    "default-param-last": "off",
+    "one-var": "off",
+    "one-var-declaration-per-line": "off"
+  },
+  "ignorePatterns": [
```
Comment (mikenath223) [inline .eslintrc.json]: - [x] You have added way more eslint rules than listed, kindly use the same eslint rules as stated [here](https://github.com/microverseinc/linters-config/tree/master/react-redux).

File: README.md [hunk 2]
```diff
@@ -1,70 +1,53 @@
...
+Codate is an application that gives insight into cities around the world. It enables uses to get understand the best place to live and work based on the cost of living, health care, internet access and some other relevant information. The application uses data from the Teleport Api and makes a meaningful display of the data.
 
-This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+Cities List View                                  |  Single City View                              |
+:------------------------------------------------:|:-----------------------------------------------:
+![](./src/assets/screenshot1.png)                 |  ![](./src/assets/screenshot2.png)             | 
 
-## Available Scripts
+## Online versions
+- [Heroku Deployment](https://codate-app.herokuapp.com/)
```
Comment (mikenath223) [inline README.md:9]: - The UI looks pretty good!

File: src/__tests__/CitiesList.test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    const payload = {
+      id: 4,
+      name: 'Cairo',
+      population: '12,000,000',
+      country: 'Egypt',
+      timeZone: 'Africa',
+      moreInfo: {
+        cityImage: 'test.png',
+      },
+    };
```
Comment (mikenath223) [inline src/__tests__/CitiesList.test.js:21]: - [x] You only added test for mock utils. kindly add component tests using react testing library and jest.

File: README.md (hunk 2 above)
Comment (mmsesay) [reply README.md:9]: Thank you.

File: src/__tests__/CitiesList.test.js (hunk 3 above)
Comment (mmsesay) [reply src/__tests__/CitiesList.test.js:21]: I have added the test for the components.

File: .eslintrc.json (hunk 1 above)
Comment (mmsesay) [reply .eslintrc.json]: Thanks for the review, I have removed them.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Muhammad, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 3 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**